### PR TITLE
Defaults to Java + Ability to configure Java version.

### PR DESCRIPTION
**Pull request by corneliouzbett**
This PR adds the ability to configure Java version.

### CODE REVIEW

**Comment by corneliouzbett** [comment]
@enyachoke @rbuisson This change will  create a cascading impact on the repositories that are currently dependent on it. Is the project versioned? Is there a way of testing this change locally?

File: .github/workflows/build-publish-workflow.yml [hunk 1]
```diff
@@ -29,11 +32,11 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Set up JDK 8
+      - name: Set up JDK ${{ inputs.java-version }}
```
Comment (mks-d) [inline .github/workflows/build-publish-workflow.yml:35]: >This change will create a cascading impact on the repositories that are currently dependent on it.

Wouldn't there be a way to default to `8` is the version isn't provided explicitly?

Comment (corneliouzbett) [reply .github/workflows/build-publish-workflow.yml:35]: Yeah there's one, not sure, I haven't used before. Something like;
```yml
with:
    distribution: 'temurin'
     java-version: ${{ inputs.java-version || '8' }}
```

**Review by mks-d — APPROVED** [review]
LGTM.

**Review by enyachoke — APPROVED** [review]

**Pull request merged**